core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -1981,6 +1998,17 @@ private[zio] trait ZIOFunctions extends Serializable {
   final def sequencePar[R >: LowerR, E <: UpperE, A](as: Iterable[ZIO[R, E, A]]): ZIO[R, E, List[A]] =
     collectAllPar[R, E, A](as)
 
+  /**
+   * Evaluate each effect in the structure with `collectAllPar`, and collect
+   * the results with given partial function.
+   */
+  final def collectWithPar[R >: LowerR, E <: UpperE, A, U](
```

Comment:
We should also add `collectAllSuccessesPar`, for consistency, and probably `ParN` as well.